src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -465,8 +465,8 @@ protected void setupChild(Configuration childConf)
         try {
             child = moduleFactory.createModule(name);
 
-            if (child instanceof AutomaticBean) {
-                final AutomaticBean bean = (AutomaticBean) child;
+            if (child instanceof AbstractAutomaticBean) {
+                final AbstractAutomaticBean bean = (AbstractAutomaticBean) child;
```

Comment:
We can not change all existing numerous third-party Checks.
It will take time before they migrate, we need to support both classes during load of Checks.